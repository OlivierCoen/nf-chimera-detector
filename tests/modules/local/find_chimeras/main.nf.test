nextflow_process {

    name "Test Process FIND_CHIMERAS"
    script "modules/local/find_chimeras/main.nf"
    process "FIND_CHIMERAS"
    tag "find_chimeras"

    test("1 chimera found") {

        when {
            process {
                """
                input[0] = [
                    [ taxid: 1234, id: 5678 ],
                     file( "${projectDir}/tests/test_data/blast_outputs/chimeras/blastOnTEs.txt", checkIfExists: true),
                     file( "${projectDir}/tests/test_data/blast_outputs/chimeras/blastOnVirusGenome.txt", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("No chimera found") {

        when {
            process {
                """
                input[0] = [
                    [ taxid: 1234, id: 5678 ],
                     file( "${projectDir}/tests/test_data/blast_outputs/no_chimeras/blastOnTEs.txt", checkIfExists: true),
                     file( "${projectDir}/tests/test_data/blast_outputs/no_chimeras/blastOnVirusGenome.txt", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("One file is empty") {

        when {
            process {
                """
                input[0] = [
                    [ taxid: 1234, id: 5678 ],
                     file( "${projectDir}/tests/test_data/blast_outputs/empty/blastOnTEs.txt", checkIfExists: true),
                     file( "${projectDir}/tests/test_data/blast_outputs/empty/blastOnVirusGenome.txt", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
