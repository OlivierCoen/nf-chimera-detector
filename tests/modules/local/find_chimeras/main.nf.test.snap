{
    "No chimera found": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        "FIND_CHIMERAS",
                        "R",
                        "4.2.1 (2022-06-23)"
                    ]
                ],
                "2": [
                    [
                        "FIND_CHIMERAS",
                        "data.table",
                        "1.15.4"
                    ]
                ],
                "csv": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T15:30:50.033908881"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "taxid": 1234,
                            "id": 5678
                        },
                        "5678_chimeras.csv:md5,d39c35b0e6dc831d0a84cf9225d736c7"
                    ]
                ],
                "1": [
                    [
                        "FIND_CHIMERAS",
                        "R",
                        "4.2.1 (2022-06-23)"
                    ]
                ],
                "2": [
                    [
                        "FIND_CHIMERAS",
                        "data.table",
                        "1.15.4"
                    ]
                ],
                "chimeras": [
                    [
                        {
                            "taxid": 1234,
                            "id": 5678
                        },
                        "5678_chimeras.csv:md5,d39c35b0e6dc831d0a84cf9225d736c7"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T08:49:57.866493083"
    },
    "1 chimera found": {
        "content": [
            {
                "0": [
                    
                ],
                "csv": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T15:58:19.167165596"
    }
}